### PR TITLE
community: add Couchbase Vector Store

### PR DESCRIPTION
**Pull request by nithishr**
- **Description:** Added support for Couchbase Vector Search to LangChain.
- **Dependencies:** couchbase>=4.1.12
- **Twitter handle:** @nithishr

### CODE REVIEW

**Comment by nithishr** [comment]
@baskaryan @efriis Could you please review this PR? I can share access to a Couchbase instance if required.

File: libs/community/tests/unit_tests/vectorstores/test_indexing_docs.py [hunk 1]
```diff
@@ -55,6 +55,7 @@ def check_compatibility(vector_store: VectorStore) -> bool:
         "BigQueryVectorSearch",
         "Cassandra",
         "Chroma",
+        "CouchbaseVectorStore",
```
Comment (baskaryan) [inline libs/community/tests/unit_tests/vectorstores/test_indexing_docs.py:58]: we shoudl update the docs/docs/modules/data_connection/index.ipynb page to list this vector store

**Review by baskaryan — COMMENTED** [review]
lgtm other than one doc addition

File: libs/community/tests/unit_tests/vectorstores/test_indexing_docs.py (hunk 1 above)
Comment (nithishr) [reply libs/community/tests/unit_tests/vectorstores/test_indexing_docs.py:58]: Thanks, did not see that. I have added it now.

**Comment by nithishr** [comment]
@baskaryan Could you please review this PR again? I have added the documentation that you requested.

File: libs/community/langchain_community/vectorstores/__init__.py [hunk 2]
```diff
@@ -109,6 +110,12 @@
 }
 
 
+def _import_couchbase() -> Any:
```
Comment (hwchase17) [inline libs/community/langchain_community/vectorstores/__init__.py:113]: where is this used?

Comment (nithishr) [reply libs/community/langchain_community/vectorstores/__init__.py]: Sorry, it was from debugging. It is not used anywhere. I have removed it now. Could you please review again?

**Review by hwchase17 — APPROVED** [review]

**Pull request merged**